### PR TITLE
[4/n][pipeline-gen] Define PipelineGeneratorConfig

### PR DESCRIPTION
**Pull request by khluu**
Define `PipelineGeneratorConfig` class to use for `PipelineGenerator`

### CODE REVIEW

File: scripts/pipeline_generator/pipeline_generator.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @property
+    def container_image(self) -> str:
+        return f"{self.container_registry}/{self.container_registry_repo}:{self.commit}"
+
+    @field_validator("test_path")
+    @classmethod
+    def check_test_path(cls, value: str) -> str:
+        if not os.path.exists(value):
+            raise ValueError(f"Test path {value} does not exist")
```
Comment (aslonnie) [inline scripts/pipeline_generator/pipeline_generator.py:26]: field validator is supposed to be checking if the values of the fields are "legal", and they are supposed to be pure functions. it is part of the marshal / unmarshaling operation of a data structure, so it cannot have dependencies on the file system.

for example, if one marshals an object into bytes, it should always be able to unmarshal the bytes back to the object, no matter what the state of the file system is (or if it even has a file system).

please check against the file system in another explicit function.

Comment (khluu) [reply scripts/pipeline_generator/pipeline_generator.py]: I removed these validators for now.. will add them into helper later on

File: scripts/tests/pipeline_generator/test_pipeline_generator.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        )
+        assert pipeline_generator_config.container_image == f"{container_registry}/{container_registry_repo}:{TEST_COMMIT}"
+
+
+def test_get_pipeline_generator_config_invalid_commit():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        with open(os.path.join(temp_dir, TEST_FILE_PATH), "w") as f:
+            f.write("content")
+        with open(os.path.join(temp_dir, EXTERNAL_HARDWARE_TEST_FILE_PATH), "w") as f:
+            f.write("content")
```
Comment (aslonnie) [inline scripts/tests/pipeline_generator/test_pipeline_generator.py:40]: you do not need these test file setup now?

Comment (khluu) [reply scripts/tests/pipeline_generator/test_pipeline_generator.py]: removed

**Review by aslonnie — COMMENTED** [review]
discussed offline. please drop pydantic as this is an internal object rather than for marshaling.

**Comment by khluu** [comment]
Updated with built in Python class and separate `validate` method

**Review by aslonnie — APPROVED** [review]

**Pull request merged**